src/Microsoft.TemplateEngine.Abstractions/Constraints/IWorkloadsInfoProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Abstractions.Constraints
+{
+    /// <summary>
+    /// Provider of descriptors of SDK workloads available to particular host (that is usually providing this component).
+    /// </summary>
+    public interface IWorkloadsInfoProvider : IIdentifiedComponent
+    {
+        /// <summary>
+        /// Set of installed workloads.
+        /// </summary>
+        public IEnumerable<WorkloadInfo> InstalledWorkloads { get; }
```

Comment:
How will you be implementing this in the CLI host? Does the Template Locator have an API surface that we can call from VS to pass this to the template engine?  Currently VS knows nothing about installed SDK workloads, we only grab the list of packages from installed workloads from the template locator.

Reply:
I'm going to open PR in SDK dueing today/tomorrw and I'll link it here.

SDK has [`VisualStudioWorkloads.GetInstalledWorkloads`](https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/commands/dotnet-workload/list/VisualStudioWorkloads.cs#L60) that does this (and I'll be utilizing it in my helper) - however it is used only for Windows platform. I was hoping to get from you some knowledge or PoCs with knowledge about situation on Mac.

Another question is how to handle VS instances with other older SDK versions: 
a) backport the utility code (not sure if it's a good idea at all)
b) inject custom provider that utilizes `sdk workload list` or any available API
c) not support the Workload constraints (same will happen by simply not passig the provider from host)

Let's have a discussion about this topic